pkg/services/cloudmigration/cloudmigrationimpl/cloudmigration.go

```diff
@@ -227,9 +250,130 @@ func (s *Service) UpdateMigration(ctx context.Context, id int64, cm cloudmigrati
 	return nil, nil
 }
 
-func (s *Service) RunMigration(ctx context.Context, uid string) (*cloudmigration.CloudMigrationRun, error) {
-	// TODO: Implement method
-	return nil, nil
+func (s *Service) GetMigrationDataJSON(ctx context.Context, id int64) ([]byte, error) {
+	var migrationDataSlice []cloudmigration.MigrateDataRequestItemDTO
+	// Data sources
+	dataSources, err := s.getDataSources(ctx, id)
+	if err != nil {
+		s.log.Error("Failed to get datasources", "err", err)
+		return nil, err
+	}
+	migrationDataSlice = append(migrationDataSlice, cloudmigration.MigrateDataRequestItemDTO{
+		Type:  cloudmigration.DatasourceDataType,
+		RefID: strconv.Itoa(int(id)),
+		Name:  "datasources",
+		Data:  dataSources,
+	})
+	// Dashboards
+	dashboards, err := s.getDashboards(ctx, id)
+	if err != nil {
+		s.log.Error("Failed to get dashboards", "err", err)
+		return nil, err
+	}
+	migrationDataSlice = append(migrationDataSlice, cloudmigration.MigrateDataRequestItemDTO{
+		Type:  cloudmigration.DashboardDataType,
+		RefID: strconv.Itoa(int(id)),
+		Name:  "dashboards",
+		Data:  dashboards,
+	})
+	// Folders
+	folders, err := s.getFolders(ctx, id)
+	if err != nil {
+		s.log.Error("Failed to get folders", "err", err)
+		return nil, err
+	}
+	migrationDataSlice = append(migrationDataSlice, cloudmigration.MigrateDataRequestItemDTO{
+		Type:  cloudmigration.FolderDataType,
+		RefID: strconv.Itoa(int(id)),
+		Name:  "folders",
+		Data:  folders,
+	})
+	migrationData := cloudmigration.MigrateDataRequestDTO{
+		Items: migrationDataSlice,
+	}
+	result, err := json.Marshal(migrationData)
+	if err != nil {
+		s.log.Error("Failed to marshal datasources", "err", err)
+		return nil, err
+	}
+	return result, nil
+}
+
+func (s *Service) getDataSources(ctx context.Context, id int64) ([]datasources.AddDataSourceCommand, error) {
+	dataSources, err := s.dsService.GetAllDataSources(ctx, &datasources.GetAllDataSourcesQuery{})
+	if err != nil {
+		s.log.Error("Failed to get all datasources", "err", err)
+		return nil, err
+	}
+
+	result := []datasources.AddDataSourceCommand{}
```

Comment:
You could make your return type for this function a slice of `MigrateDataRequestItemDTO`. Then you don't have to do much work in the other function.